src/sentry/incidents/logic.py

```diff
@@ -540,6 +540,91 @@ def create_alert_rule(
     return alert_rule
 
 
+def update_alert_rule(
```

Comment:
Should this method be run inside a database transaction?

Reply:
Not sure it's necessary? We only do one database update here. Worst case is we fail on deleting the old subscription id and don't rollback. But if we do rollback then the new subscription id won't be deleted.

S&S has a plan to deal with orphaned subscriptions, so maybe not a huge deal?